Library/Homebrew/os/mac.rb

```diff
@@ -79,19 +80,25 @@ def active_developer_dir
       @active_developer_dir ||= Utils.popen_read("/usr/bin/xcode-select", "-print-path").strip
     end
 
-    def sdk_path(v = version)
-      (@sdk_path ||= {}).fetch(v.to_s) do |key|
-        opts = []
-        # First query Xcode itself
-        opts << Utils.popen_read(locate("xcodebuild"), "-version", "-sdk", "macosx#{v}", "Path").chomp
-        # Xcode.prefix is pretty smart, so lets look inside to find the sdk
-        opts << "#{Xcode.prefix}/Platforms/MacOSX.platform/Developer/SDKs/MacOSX#{v}.sdk"
-        # Xcode < 4.3 style
-        opts << "/Developer/SDKs/MacOSX#{v}.sdk"
-        @sdk_path[key] = opts.map { |a| Pathname.new(a) }.detect(&:directory?)
+    # Returns the requested SDK, if installed.
+    # If the requested SDK is not installed returns either:
+    # a) The newest SDK (if any SDKs are available), or
+    # b) nil
+    def sdk(v = version)
+      @locator ||= SDKLocator.new
+      begin
+        @locator.sdk_for v
+      rescue SDKLocator::NoSDKError
+        @locator.latest_sdk
       end
     end
 
+    # Returns the path to an SDK or nil, following the rules set by #sdk.
+    def sdk_path(v = version)
+      s = sdk(v)
+      s.path unless s.nil?
```

Comment:
I tend to prefer avoiding the `if (foo = bar)` construction unless it's in a conditional with many branches, since I don't personally think it reads quite as clearly.